File: MultiplesOf3Or5.m
#import <Foundation/Foundation.h>

int sumOfMultiplesOf(int v1, int v2, int range) {
   
    int sum = 0;
    int temp = 0;
    int counter = 0;
    
    for (temp=0; temp<=range-2; temp++) {
        counter++;
        if(counter % v1 == 0 || counter % v2 == 0) {
            sum += counter;
        }
    }
    
    return sum;
}

int main(int argc, const char * argv[]) {
    
    int sum = 0;
    
    @autoreleasepool {
        sum = sumOfMultiplesOf(3,5,1000);
        printf("%d\n", sum);
    }
    return 0;
}

File: MultiplesOf3Or5.swift
// Project Euler - Multiples of 3 or 5

func multiplesOf(first valueOne: Int, second valueTwo: Int, range: Int) {
    let v1 = valueOne
    let v2 = valueTwo
    var sum: Int = 0

    var counter = 1
    
    for _ in 0...range-3 {
        counter += 1
        if counter % v1 == 0 {
            sum += counter
        }
        
        if counter % v2 == 0 {
            sum += counter
        }
    }
    
    print(sum)
}
